pallets/stablewap/src/benchmarks.rs

```diff
@@ -0,0 +1,227 @@
+// This file is part of Basilisk-node
+
+// Copyright (C) 2020-2022  Intergalactic, Limited (GIB).
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use super::*;
+
+use frame_benchmarking::account;
+use frame_benchmarking::benchmarks;
+use frame_support::pallet_prelude::DispatchError;
+use frame_system::RawOrigin;
+use orml_traits::MultiCurrency;
+use orml_traits::MultiCurrencyExtended;
+use sp_runtime::Permill;
+
+use hydradx_traits::Registry;
+
+use crate::types::{Balance, PoolId};
+
+benchmarks! {
+	 where_clause {  where T::AssetId: From<u32> + Into<u32>,
+		T::Currency: MultiCurrencyExtended<T::AccountId, Amount=i128>,
+		T::AssetRegistry: Registry<T::AssetId, Vec<u8>, Balance, DispatchError>,
+		T: crate::pallet::Config
+	}
+
+	create_pool{
+		let token_a = T::AssetRegistry::create_asset(&b"one".to_vec(), 1u128)?;
+		let token_b = T::AssetRegistry::create_asset(&b"two".to_vec(), 1u128)?;
+
+		// Note: this is extreme case where calculate_d does around 40 iterations.
```

Comment:
The limit is 128, though, right?
Is this the max possible?

Reply:
this was already changed. 

math does always max iterations in benchmarks.

Reply:
Resovled by the math crate always doing max iterations on `runtime_benchmarks`